src/vm/jithelpers.cpp

```diff
@@ -2369,7 +2378,12 @@ BOOL ObjIsInstanceOf(Object *pObject, TypeHandle toTypeHnd, BOOL throwCastExcept
         }
         GCPROTECT_END(); //exception
     }
-#endif // FEATURE_ICASTABLE
+#endif // FEATURE_ICASTABLE   
+
+    if (canCache)
+    {
+        CastCache::TryAddToCache(fromTypeHnd, toTypeHnd, fCast);
```

Comment:
If we have the CastCache perhaps we would always be able to use tagged pointers to indicate if caching is possible however a CastCache is more useful by virtue of that statement.